Atomic/syscontainers.py

```diff
@@ -2129,84 +2125,47 @@ def get_layers_from_manifest(manifest):
             layers = manifest.get("Layers")
         return layers
 
-    def _import_layers_into_ostree(self, repo, imagebranch, manifest, layers):
-        def get_directory_size(path):
-            size = 0
-            seen = {}
-            for root, _, files in os.walk(path):
-                for f in files:
-                    s = os.lstat(os.path.join(root, f))
-                    key = "%s-%s" % (s.st_dev, s.st_ino)
-                    if key not in seen:
-                        seen[key] = key
-                        size += s.st_size
-            return GLib.Variant('s', str(size))
-
-        repo.prepare_transaction()
-        for layer, tar in layers.items():
-            mtree = OSTree.MutableTree()
-            def filter_func(*args):
-                info = args[2]
-
-                if info.get_file_type() == Gio.FileType.SPECIAL:
-                    return OSTree.RepoCommitFilterResult.SKIP
-
-                if info.get_file_type() == Gio.FileType.DIRECTORY:
-                    info.set_attribute_uint32("unix::mode", info.get_attribute_uint32("unix::mode") | stat.S_IWUSR)
-                return OSTree.RepoCommitFilterResult.ALLOW
-
-            modifier = OSTree.RepoCommitModifier.new(0, filter_func, None)
-
-            checkout = self._get_system_checkout_path()
-            destdir = checkout if os.path.exists(checkout) else None
-
-            try:
-                temp_dir = tempfile.mkdtemp(prefix=".", dir=destdir)
-                # NOTE: tarfile has an issue with utf8. This works around the problem
-                # by using the systems tar command.
-                # Ref: https://bugzilla.redhat.com/show_bug.cgi?id=1194473
-                subprocess.check_call(['tar', '-xf', tar, '-C', temp_dir])
-                if self.user:
-                    SystemContainers._correct_dir_permissions_for_user(temp_dir)
-
-                repo.write_directory_to_mtree(Gio.File.new_for_path(temp_dir), mtree, modifier)
-                root = repo.write_mtree(mtree)[1]
-
-                metav = GLib.Variant("a{sv}", {'docker.layer': GLib.Variant('s', layer),
-                                               'docker.size': get_directory_size(temp_dir)})
-                csum = repo.write_commit(None, "", None, metav, root)[1]
-            finally:
-                shutil.rmtree(temp_dir, ignore_errors=True)
+    def _pull_docker_image(self, image):
```

Comment:
Please add docstring. Example:

```python
def _pull_docker_image(self, image):
    """
    pulls a docker image down to a local tar archive.

    :param image: The full name of the image to pull down.
    :type image: str
    :returns: What does this return?
    :rtype: What's the type?
    """
    ...
```

Reply:
yup, sure, I kinda got lazy when the method itself is short. :P

Reply:
Ha! I understand. One of the things I'm trying to push in ``atomic`` is to get more code docs so even documenting the small ones helps others who jump in later understand what's going on.